lib/streamlit/runtime/connection_factory.py

```diff
@@ -171,25 +171,52 @@ def connection_factory(
 
     Config options, credentials, secrets, etc. for connections are sourced from various
     sources:
```

Comment:
The first line (short description)

> Create a new connection or return an existing one.

should specify connection to what (perhaps data sources or APIs?).  There could optionally also be a second line (long description) below briefly elaborating on  what st.connection does.